scipy/stats/tests/test_quantile.py

```diff
@@ -95,21 +103,46 @@ def test_against_numpy(self, method, shape_x, shape_p, axis, xp):
 
     @pytest.mark.parametrize('axis', [0, 1])
     @pytest.mark.parametrize('keepdims', [False, True])
-    @pytest.mark.parametrize('nan_policy', ['omit', 'propagate'])
+    @pytest.mark.parametrize('nan_policy', ['omit', 'propagate', 'marray'])
     @pytest.mark.parametrize('dtype', ['float32', 'float64'])
-    def test_against_reference(self, axis, keepdims, nan_policy, dtype, xp):
+    @pytest.mark.parametrize('method', ['linear', 'harrell-davis'])
+    def test_against_reference(self, axis, keepdims, nan_policy, dtype, method, xp):
         rng = np.random.default_rng(23458924568734956)
-        shape = (3, 4)
+        shape = (5, 6)
         x = rng.random(size=shape).astype(dtype)
         p = rng.random(size=shape).astype(dtype)
-        mask = rng.random(size=shape) > 0.9
+        mask = rng.random(size=shape) > 0.8
         assert np.any(mask)
         x[mask] = np.nan
         if not keepdims:
             p = np.mean(p, axis=axis, keepdims=True)
 
+        # inject p = 0 and p = 1 to test edge cases
+        # Currently would fail with CuPy/JAX (cupy/cupy#8934, jax-ml/jax#21900)
+        if is_numpy(xp):
+            p0 = p.ravel()
+            p0[1] = 0.
+            p0[-2] = 1.
+
         dtype = getattr(xp, dtype)
-        kwargs = dict(axis=axis, keepdims=keepdims, nan_policy=nan_policy)
+
+        if nan_policy == 'marray':
+            if method == 'harrell-davis':
+                pytest.skip("Needs gh-22490")
+            if is_torch(xp):
+                pytest.skip("sum_cpu not implemented for UInt64, see "
+                            "data-apis/array-api-compat#242")
+            marray = pytest.importorskip('marray')
```

Comment:
Had to google it. So this either imports `marray` or skips the test if `marray` is not available, correct? It'd be way more readable IMO to use a "straitghforward" way with a top-level import and a module-level variable.
```
try:
   import marray
   HAVE_MARRAY = True
except ModuleNotFoundError:
   HAVE_MARRAY=False
```

Reply:
Yes, that's what it does.

In fact I had a `try`-`except` for this purpose in the original `marray` PR, but it was recommended I use `importorskip` instead.

I didn't know about it either, but it seems extremely handy to me. Why have all that at the top, then two lines of `if not HAVE_MARRAY: pytest.skip()` here when it can all just be one line?

My guess is that the second or third time you encounter it, you'll find it more readable than the alternative.

Reply:
Well, my concern is obviously not about readability. I'm concerned about hiding optional dependencies  deep in test functions. 
That said, if you as a stats maintainer find it good enough, great,  it's not worth splitting hairs over.